src/main.rs

```diff
@@ -90,8 +100,14 @@ fn main() -> sysexits::ExitCode {
             return sysexits::ExitCode::Config;
         }
     };
-    if matches.opt_present("help") || (!matches.opt_present("git") && !matches.opt_present("input"))
-    {
+    let input = match commit_analyzer::InputMethod::parse(&matches) {
+        Some(input) => input,
+        None => {
+            eprintln!("{}", opts.usage("Please specify one input method."));
```

Comment:
This is also an elegant solution:  one error message for two cases.  The meaning depends on the calling situation but this way, the cases no input method / multiple input methods can be combined most efficiently.